gpii/node_modules/eventLog/src/eventLog.js

```diff
@@ -0,0 +1,286 @@
+
+/**
+ * Logs an error caught by onUncaughtException or failureEvent.
+ *
+ * @param err The error.
+ */
+gpii.eventLog.gotError = function (err) {
+    var eventLog = fluid.queryIoCSelector(fluid.rootComponent, "gpii.eventLog");
+    if (eventLog.length > 0) {
+        eventLog[0].logError(eventLog[0], null, "Failure", err);
+    }
+};
+
+/**
+ * Gets the path of the new log file for this instance of gpii.
+ *
+ * @param that {Component} The gpii.eventLog instance.
+ */
+gpii.eventLog.getLogFile = function (that) {
+    var logFile = that.logFilePath || process.env.GPII_EVENT_LOG;
+
+    if (!logFile) {
+        var startupTime = Date.now();
+        var gpiiSettingsDir = that.getGpiiSettingsDir();
+        logFile = path.join(gpiiSettingsDir, "gpii-" + gpii.journal.formatTimestamp(startupTime) + ".log");
+        fluid.log(fluid.logLevel.IMPORTANT, "Writing event log to " + that.logFilePath);
+    }
+
+    that.logFilePath = logFile;
+
+    return that.logFilePath;
+};
+
+/**
+ * Writes an event to the log file.
+ *
+ * @param that {Component} The gpii.eventLog instance.
+ * @param level {Object} Level of the log, see fluid.logLevelsSpec [FATAL,FAIL,WARN,IMPORTANT,INFO,TRACE].
+ * @param event {Object} The object. This will be modified to what has been sent to the log.
+ */
+gpii.eventLog.writeLog = function (that, level, event) {
+    var intLevel = gpii.eventLog.checkLevel(level);
+    event.level = intLevel.value;
+
+    // Log to console before the installation ID and timestamp are added (no one wants to see it).
+    fluid.log(fluid.logLevel.IMPORTANT, event);
+
+    event.installID = that.installationID;
+    event.timestamp = gpii.eventLog.getTimestamp();
+    fs.appendFileSync(that.logFilePath, JSON.stringify(event) + "\n");
+};
+
+/**
+ * Ensures a value is an object. If it's not an object, return an object containing it.
+ * Null, empty object, or undefined returns null.
+ *
+ * @param data The value to work with.
+ * @return {object} The given object, or new object with a field "value" set to data
+ */
+gpii.eventLog.wrapData = function (data) {
+    var dataTogo;
+
+    if (data === null || data === undefined) {
+        dataTogo = null;
+    } else if (fluid.isPlainObject(data, true)) {
+        dataTogo = $.isEmptyObject(data) ? null : data;
+    } else {
+        dataTogo = {value: data};
+    }
+
+    return dataTogo;
+};
+
+/**
+ * Ensure that the loglevel has a valid value. The levels are defined in the fluid.logLevelsSpec
+ * Sets INFO as default loglevel
+ *
+ * @param level to check, can be a string that represents the value or a property of fluid.logLevel
+ * @return a valid fluid.logLevel, with INFO as default
+ */
+gpii.eventLog.checkLevel = function (level) {
+    if (typeof level === "string" && level in fluid.logLevelsSpec) {
+        return fluid.logLevel[level];
+    } else {
+        return fluid.isLogLevel(level) ? level : fluid.logLevel.INFO;
+    }
+};
+
+fluid.defaults("gpii.eventLog.log", {
```

Comment:
I don't think anything will use this definition? In general these are either i) model transformation functions or ii) things you might want to invoke with named arguments via fluid.invokeGradedFunction